Snippets/Core/Core_6/UpgradeGuides/5to6/UpgradeMessageHandler.cs

```diff
@@ -37,4 +37,88 @@ await context.Publish(new MyEvent())
         }
     }
     #endregion
+
+
+    #region 5to6-handler-migration-beginning
+    public class MigrationBeginning : IHandleMessagesFromPreviousVersions<MyMessage>
+    {
+        public IBus Bus { get; set; }
+
+        public void Handle(MyMessage message)
+        {
+            Bus.Send(new MyOtherMessage());
+            Bus.Publish(new MyEvent());
+        }
+    }
+    #endregion
+
+    #region 5to6-handler-migration-step1
+    public class MigrationStep1 : IHandleMessages<MyMessage>
+    {
+        public IBus Bus { get; set; }
+
+        public void Handle(MyMessage message)
+        {
+            Bus.Send(new MyOtherMessage());
+            Bus.Publish(new MyEvent());
+        }
+
+        public async Task Handle(MyMessage message, IMessageHandlerContext context)
+        {
+        }
+    }
+    #endregion
+
+    #region 5to6-handler-migration-step2
+    public class MigrationStep2 : IHandleMessages<MyMessage>
+    {
+        public IBus context { get; set; }
```

Comment:
Because then and only then you can just inline the code since message
handler context param name is called context

On Tuesday, June 28, 2016, Indu Alagarsamy <notifications@github.com> wrote:

> In Snippets/Core/Core_6/UpgradeGuides/5to6/UpgradeMessageHandler.cs
> <https://github.com/Particular/docs.particular.net/pull/1642#discussion_r68813945>
> :
>
> > +        public void Handle(MyMessage message)
> > +        {
> > +            Bus.Send(new MyOtherMessage());
> > +            Bus.Publish(new MyEvent());
> > +        }
> > +
> > +        public async Task Handle(MyMessage message, IMessageHandlerContext context)
> > +        {
> > +        }
> > +    }
> > +    #endregion
> > +
> > +    #region 5to6-handler-migration-step2
> > +    public class MigrationStep2 : IHandleMessages<MyMessage>
> > +    {
> > +        public IBus context { get; set; }
>
> What's the purpose of renaming Bus to context variable?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Particular/docs.particular.net/pull/1642/files/cb3b63006af3e1ed67464dfd0f4af1820aa83cc0#r68813945>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAKosuKrMx9xdyccGk_jwKiPiiGdB8gcks5qQWbegaJpZM4JACq2>
> .
>